### PR TITLE
CD-137: Linux- Monitor IP Latency from a host to another using Domotz

### PR DESCRIPTION
**Pull request by mariemm97**

### CODE REVIEW

**Review by mohamedkallel — APPROVED** [review]

File: examples/ssh/linux_iplatency.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
+/**
+ * Domotz Custom Driver 
+ * Name: Linux Monitor IP Latency
+ * Description: This script is designed to ping an IP address and retrieve the average latency and packet loss percentage.
```
Comment (GeorgiTsatsev) [inline examples/ssh/linux_iplatency.js:4]: its designed to ping a list of IP addresses from a linux host machine

File: examples/ssh/linux_iplatency.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+ *      - Ubuntu 20.04 LTS
+ * Shell Version:
+ *      - Bash 5.1.16
+ * 
+ * Creates a Custom Driver table with a list of IP addresses, their average latency, and their packet loss.
+ * 
+ **/
+
+var pktno = "2"; // Number of packets to send during the ping command.
+var ipAddresses = ["8.8.8.8", "1.1.1.1", "192.168.0.1", "192.168.0.1", "172.17.0.1"]; // List of IP addresses to ping and retrieve status for.
```
Comment (GeorgiTsatsev) [inline examples/ssh/linux_iplatency.js:18]: Please leave just two entries.

Also, you have duplicated IP address and this crashes the current version execution

File: examples/ssh/linux_iplatency.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    D.device.sendSSHCommand(sshCommandOptions, function (output, error) {
+        if(error) checkSshError(error);
+        d.resolve(output);
+    });
+    return d.promise;
+}
+
+/**
+ * Excuting a simple command to test access to device:
+ * 'ls' list the files and directories within a specified directory
```
Comment (GeorgiTsatsev) [inline examples/ssh/linux_iplatency.js:54]: Do not use 'ls' for validation

Please properly check if "ping" command is supported on the host device with the needed parameters

File: examples/ssh/linux_iplatency.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+ * @remote_procedure
+ * @label Validate Association
+ * @documentation This procedure is used to validate if the driver can be applied on a device during association as well as validate any credentials provided
+ */
+function validate() {
+    var commandValidate = "ls";
+    console.info("Verifying credentials ... ", commandValidate); 
+    executeCommand(commandValidate)
+        .then(function(){
+            D.success();
```
Comment (GeorgiTsatsev) [inline examples/ssh/linux_iplatency.js:64]: You are not considering failure cases where the command is not recognized

File: examples/ssh/linux_iplatency.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        .then(function () {
+            D.success(tableColumns);
+        })
+        .catch(function (error) {
+            console.error(error);
+            D.failure(D.errorType.GENERIC_ERROR);
+        });
+}
+
+function parseOutput(output, ipAddress){
```
Comment (GeorgiTsatsev) [inline examples/ssh/linux_iplatency.js:100]: You are not handling error cases for a single address (for example 100% packet loss)

{"code":1,"output":"PING 8.8.8.33 (8.8.8.33) 56(84) bytes of data.\n\n--- 8.8.8.33 ping statistics ---\n2 packets transmitted, 0 received, 100% packet loss, time 1003ms"}

File: examples/ssh/linux_iplatency.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
+{
+    "logo": "linux.svg",
+    "version": "1.0.0",
+    "name": "Linux Monitor IP Latency",
+    "description": "Monitor IP Latency from a host to another",
```
Comment (GeorgiTsatsev) [inline examples/ssh/linux_iplatency.json:5]: to a list of IP addresses

File: examples/ssh/linux_iplatency.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        "sandbox_version": "1.12",
+        "others": [
+            "Linux",
+            "SSH Enabled",
+            "ping"
+        ]
+    },
+    "tags": [
+        "linux",
+        "ip address",
```
Comment (GeorgiTsatsev) [inline examples/ssh/linux_iplatency.json:18]: Tags do not support empty spaces, replace with '-'

File: examples/ssh/linux_iplatency.json [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        "latency",
+        "packet loss"
+    ],
+    "sample_period": 30000,
```
Comment (GeorgiTsatsev) [inline examples/ssh/linux_iplatency.json:22]: the sample period is not OK (its in seconds). Change to 1800

**Review by GeorgiTsatsev — APPROVED** [review]

**Pull request merged**